From: Modestas Vainius <modestas@vainius.eu>
Subject: Attempt to port to gpsd 2.90 API. Might not work, not tested
 Based on http://gpsd.berlios.de/protocol-transition.html
          http://gpsd.berlios.de/gpsd.html
          http://gpsd.berlios.de/libgps.html
 Probably should be replaced by proper fix from upstream (when available).
Forwarded: no

--- a/marble/src/plugins/positionprovider/gpsd/GpsdConnection.cpp
+++ b/marble/src/plugins/positionprovider/gpsd/GpsdConnection.cpp
@@ -22,6 +22,7 @@
 {
     gps_data_t* data = m_gpsd.open();
     if ( data ) {
+        m_gpsd.stream(WATCH_ENABLE);
         connect( &m_timer, SIGNAL( timeout() ), this, SLOT( update() ) );
         m_timer.start( 1000 );
     } else
@@ -29,7 +30,10 @@
 }
 
 void GpsdConnection::update() {
-    gps_data_t* data = m_gpsd.query( "o" );
+    gps_data_t* data;
+    while ((data = m_gpsd.poll()) && !(data->set & POLICY_SET)) {
+        data = m_gpsd.poll();
+    }
     if ( data )
         emit gpsdInfo( *data );
 }
